### DISCUSSION TITLE
Extract M and C from Kane's equation

### DISCUSSION THREAD

**Discussion by Simone2021space** [sympy/sympy#22367] [Q&A]
Hi team,
I need to derive the general dynamic equation of a manipulator : 
torque = M(q)q" + C(q,q') q' + F

starting from Kane's equation: 
fr+fr*=0

so I need to extract the M and C terms, is there a function or a tool for doing that?
If not how can I do it manually?

**Comment by moorepants** [comment]
SymPy has routines for finding coefficients, for example `.coeff()`. M is the Jacobian of fr+fr* wrt to q''. C isn't linear in q' in general, so you can't get those coefficients as you've written them.
